nums/core/array/application.py

```diff
@@ -1107,3 +1107,79 @@ def nanmean(self, a: BlockArray, axis=None, keepdims=False, dtype=None):
         if dtype is not None:
             res = res.astype(dtype)
         return res
+
+    def nanvar(self, a: BlockArray, axis=None, ddof=0, keepdims=False, dtype=None):
+        mean = self.nanmean(a, axis=axis, keepdims=True)
+        ss = self.reduce("nansum", (a - mean)**self.two, axis=axis, dtype=dtype, keepdims=keepdims)
+        num_summed = self.sum(~a.ufunc("isnan"), axis=axis, dtype=a.dtype, keepdims=keepdims) - ddof
+        res = ss / num_summed
+        if dtype is not None:
+            res = res.astype(dtype)
+        return res
+
+    def nanstd(self, a: BlockArray, axis=None, ddof=0, keepdims=False, dtype=None):
+        res = self.sqrt(self.nanvar(a, axis, ddof, keepdims))
+        if dtype is not None:
+            res = res.astype(dtype)
+        return res
+
+    def atleast_1d(self, *arys):
+        res = []
+        for ary in arys:
+            if not isinstance(ary, BlockArray):
```

Comment:
Can you look into using or creating a utility function out of `BlockArray.check_or_convert_other(self, other)` for this case? We can discuss details of this, but this behavior is creating a precedent that all operations auto-convert their inputs to BlockArray. We'll need to provide similar semantics for all operations.